activerecord/lib/active_record/relation/query_methods.rb

```diff
@@ -981,6 +995,12 @@ def build_arel(aliases)
         arel.from(build_from) unless from_clause.empty?
         arel.lock(lock_value) if lock_value
 
+        with_annotation_values.reject(&:nil?).tap do |annotations|
```

Comment:
Why reject `nil?`? Why would someone pass `nil` for that method?

Reply:
This was an artifact of iterating on the implementation, I think. Good points, thank you. I've simplified things in the latest version.